### PR TITLE
Feature - Update aws-toolkit-commons package

### PR DESCRIPTION
**Pull request by damntrecky**
- update AmazonQ Code Transform telemetry definitions for new package update

## Problem
We update the AWS Toolkit Common package and need to update the VSCode package per this PR -> https://github.com/aws/aws-toolkit-common/pull/648
## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/codewhisperer/service/transformByQHandler.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ export async function validateProjectSelection(project: vscode.QuickPickItem) {
         vscode.window.showErrorMessage(CodeWhispererConstants.noSupportedJavaProjectsFoundMessage, { modal: true })
         telemetry.codeTransform_isDoubleClickedToTriggerInvalidProject.emit({
             codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
-            codeTransformPreValidationError: 'No Java project found' as CodeTransformPreValidationError,
```
Comment (damntrecky) [inline src/codewhisperer/service/transformByQHandler.ts:96]: Typecast with `as` is unnecessary.

File: src/codewhisperer/service/transformByQHandler.ts [hunk 2]
```diff
@@ -125,8 +125,7 @@ export async function validateProjectSelection(project: vscode.QuickPickItem) {
         vscode.window.showErrorMessage(CodeWhispererConstants.noSupportedJavaProjectsFoundMessage, { modal: true })
         telemetry.codeTransform_isDoubleClickedToTriggerInvalidProject.emit({
             codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
-            codeTransformPreValidationError:
-                'Project selected is not Java 8 or Java 11' as CodeTransformPreValidationError,
+            codeTransformPreValidationError: 'UnsupportedJavaVersion',
```
Comment (damntrecky) [inline src/codewhisperer/service/transformByQHandler.ts:128]: New values DO NOT have spaces.

**Review by justinmk3 — APPROVED** [review]

**Comment by justinmk3** [comment]
https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#changelog

> Pull requests that change customer-facing behavior (in any describable way) should include a changelog item(s).

Is that the case here?

File: .changes/next-release/Bug Fix-a8d6edab-136e-4a0e-ba04-1bdd4e1a45eb.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (damntrecky) [inline .changes/next-release/Bug Fix-a8d6edab-136e-4a0e-ba04-1bdd4e1a45eb.json:1]: Should delete this since its non user facing?

Comment (justinmk3) [reply .changes/next-release/Bug Fix-a8d6edab-136e-4a0e-ba04-1bdd4e1a45eb.json:1]: yeah I'd say so. Also would suggest using "telemetry" as the commit-message prefix (aka topic)

File: src/codewhisperer/service/transformByQHandler.ts [hunk 4]
```diff
@@ -140,11 +139,18 @@ export async function validateProjectSelection(project: vscode.QuickPickItem) {
...
+                codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
+                codeTransformPreValidationError: 'NonMavenProject',
+                result: MetadataResult.Fail,
+                reason: buildType,
+            })
+        }
         telemetry.codeTransform_isDoubleClickedToTriggerInvalidProject.emit({
             codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
-            codeTransformPreValidationError: 'Only Maven projects supported' as CodeTransformPreValidationError,
+            codeTransformPreValidationError: 'NoPom',
```
Comment (justinmk3) [inline src/codewhisperer/service/transformByQHandler.ts:152]: could save some repeated code
```suggestion
            codeTransformPreValidationError: buildType === 'Gradle' ? NonMavenProject : 'NoPom',
```

**Review by leigaol — APPROVED** [review]

**Pull request merged**